sql/core/src/main/scala/org/apache/spark/sql/util/QueryExecutionListener.scala

```diff
@@ -81,7 +81,10 @@ class ExecutionListenerManager private[sql](
     loadExtensions: Boolean)
   extends Logging {
 
-  private val listenerBus = new ExecutionListenerBus(this, session)
```

Comment:
qq. can we just make this as a lazy val or a function?

Reply:
ah, the problem is in `clone` ...

Reply:
Yeah, I was worried about `clone`. Now that I think about it, though, I think a `lazy val` might be okay: most of the new `SparkSessions` will be child sessions of the root session, so the work I did in `clone` is only saving a registration of the root session and isn't saving any work in the child. Avoiding an unnecessary initialization in clone would only make a difference if you had a large number of child sessions which spawned large numbers of grandchildren sessions, but I'm not worried about that case. Let me update to make it a simple `lazy val`.

Reply:
Updated in 030f6379b3c5c6419cf6a5cb165725e88774cc3f

Reply:
I am not very familiar with the sql codebase - but I was assuming this was a common enough case ? If not, agree `lazy val` would be better here.

Reply:
Thinking about this some more, I don't think the `root -> child -> grandchild` session case will be common in Spark itself (I think the ThriftServer would only create one layer of child sessions), but that scenario might crop up in applications built on Spark. Given this, I'm going to roll back to what I had earlier with the `@volatile` Option and the initialization lock. It's a _bit_ ugly but not _that_ ugly and provides some additional benefit.